core/utils/pqueue_tag.c

```diff
@@ -27,13 +27,14 @@ static pqueue_pri_t pqueue_tag_get_priority(void* element) { return (pqueue_pri_
 
 /**
  * @brief Callback comparison function for the tag-based priority queue.
- * Return 0 if the first argument is less than second and 1 otherwise.
+ * Return -1 if the first argument is less than second, 0 if the two arguments are the same,
+ * and 1 otherwise.
  * This function is of type pqueue_cmp_pri_f.
  * @param priority1 A pointer to a pqueue_tag_element_t, cast to pqueue_pri_t.
  * @param priority2 A pointer to a pqueue_tag_element_t, cast to pqueue_pri_t.
  */
 static int pqueue_tag_compare(pqueue_pri_t priority1, pqueue_pri_t priority2) {
-  return (lf_tag_compare(((pqueue_tag_element_t*)priority1)->tag, ((pqueue_tag_element_t*)priority2)->tag) > 0);
+  return (lf_tag_compare(((pqueue_tag_element_t*)priority1)->tag, ((pqueue_tag_element_t*)priority2)->tag));
```

Comment:
I'm confused: Shouldn't a _lower_ tag have _higher_ priority?

Reply:
Yes, a _lower_ tag has to have _higher_ priority. 

In the code below, if a parent node's priority is greater than the moving node's priority, we swap them. If a moving node's index is `i`, the parent node's index is `i >> 1`. In other words, a _lower_ index element (_higher_ priority element) has a _lower_ tag.

https://github.com/lf-lang/reactor-c/blob/739b696d0728fd4cf79ea3b5749c82984d9aaf57/core/utils/pqueue_base.c#L176-L190

Also, this code aligns with the current comparison function. It also returns 1 if thiz > that.
https://github.com/lf-lang/reactor-c/blob/ec06c930a65d05420d103a6773fca8a442d58124/core/utils/pqueue.c#L16